src/lib.rs

```diff
@@ -403,6 +404,10 @@ fn get_branches(repo_path: &Path) -> Result<Vec<BranchInfo>> {
                 worktree,
             })
         })
+        .filter(|branch| match branch {
```

Comment:
```suggestion
        // This temporary branch should have been deleted but filter it out just in case something went wrong.
        .filter(|branch| !matches!(branch, Ok(b) if b.branch == TEMPORARY_BRANCH_NAME))
```

(I assume that is why you added this?)

Reply:
It seemed to always be there for me. But maybe because autorebase takes ~three hours on the repo I wanted it for, so maybe it was always getting cancelled as I was developing. :)

Reply:
Ah yeah it shouldn't be. I'll add the comment (this is the first time I've used Github PRs from the review side; didn't know reviewers can just commit suggestions themselves!)

> autorebase takes ~three hours on the repo I wanted it for,

Wow! Can you share any details? How big is the repo? How many branches are you rebasing, etc? That's pretty crazy :-)

Reply:
Oh wait that won't work because the suggestion doesn't delete enough... Hmm I'll just merge your change and add another commit later.

Reply:
I think it's mostly the commit signing, #3.

It's ~600 branches, ~10,000 committed files (`git ls-files`), and under a gig of `.git`.

I, [uh](https://xkcd.com/1172/), had a plan. Many of the branches are merged, or partly merged. I thought I could rebase them all, and then delete any which had no commits afterwards. It actually works pretty well. I'd raise a PR but it's basically just `autorebase --all-branches && git branch | xargs git branch -d`, as `git branch -d` skips branches which aren't merged.

The first attempt took out 300 branches.

Reply:
Wow that's a lot of branches! Maybe I don't want to know more... :-P